### PR TITLE
Hotfix 2.5

### PR DESCRIPTION
**Pull request by dpvc**
This fixes a number of small problems with v2.5, including #1033, #1032, #1006, #1039, and #1050.  Also, it recovers some of the speed gain in HTML-CSS that was lost in a change during the testing of v2.5 to resolve an issue there.    Finally, it fixes some problems with maction.

@pkra, are you ready to do an update to the CDN for v2.5.1?  If so, I will repack and recombine, then we can merge this and tag a new release.

I ran several broswer/OS combinations through the test suite on this branch, to make sure there were no unexpected results, and it looks good.

Did you intend the update to mml3 to be part of this hotfix?  Currently, it is in the develop branch, not the hotfix branch, so it is not in this merge right now.  If you want it to be included, I will cherry pick that commit from develop.

### CODE REVIEW

**Comment by pkra** [comment]
> Did you intend the update to mml3 to be part of this hotfix? Currently, it is in the develop branch, not the hotfix branch, so it is not in this merge right now. If you want it to be included, I will cherry pick that commit from develop.

Yes, that would be great.

**Comment by dpvc** [comment]
I have included your changes to mml3, as well as the fix for issue #1053 that was ready but hadn't been merged.  I repacked and recombined, so this is ready to go.  When you are ready to update the CDN, this can be merged into `master` and tagged as a release.  We will also need to make the corresponding pull from `hotfix-2.5` into `v2.5-latest`;  I can make a pull request for that here, if you like, or can do it by hand once this pull is made.  Once those are done, I will merge into develop by hand.

There are only a few of the files (41) that are changed, so I'm hoping you can just move those files to the CDN, rather than updating and purging everything.

This update does include a change to the HTML-CSS output that requires a change to the noErrors extension, so if the CDN updates one before the other, it might cause temporary processing errors like before.  Doing the noErrors extension first, and then the HTML-CSS jax.js file would be the order that would make it work best.  Those sites using a combined configuration file will get compatible versions automatically (since they are both in the same file), but those who roll their own configurations could experience a problem if the CDN doesn't get both new versions at the same time.  Let me know if you need more details.

**Comment by pkra** [comment]
I won't make it today. Let's do the release Monday or Tuesday.

**Comment by dpvc** [comment]
Sounds good.  Nothing critical in the hotfix.

**Comment by pkra** [comment]
Should the MathJax version be bumped to 2.5.1?

File: unpacked/MathJax.js [hunk 1]
```diff
@@ -46,8 +46,8 @@ if (window.MathJax) {window.MathJax = {AuthorConfig: window.MathJax}}
 // MathJax.isPacked = true; // This line is uncommented by the packer.
 
 MathJax.version = "2.5.0";
```
Comment (pkra) [inline unpacked/MathJax.js]: To clarify, this is what I meant with "MathJax version" in my earlier message. Should we bump this to 2.5.1?

Comment (dpvc) [inline unpacked/MathJax.js]: Yes, I guess I missed that.  I'll take care of it.

**Comment by dpvc** [comment]
There is one more thing we might want to add to this.  The change to remove measurements in HTML-CSS output (for speed) has had adverse affects when unknown characters are used (which used to be measured, but now aren't, so the widths are wrong).  This affects both unknown characters, and characters in other fonts.  See this [stack exchange post](http://stackoverflow.com/questions/28706543/mathjax-overarrow-too-short) for an example.  We could special case the unknown characters without too much trouble (I think).  Since this is a new problem with v2.5, it would fit in with a hot fix for 2.5.  I'm inclined to want to fix it if we can.

In the meantime, you could amend your answer on SE to indicate the setting `noReflows:false` in the HTML-CSS block of the configuration would fix the problem the OP was having.

**Comment by pkra** [comment]
> Since this is a new problem with v2.5, it would fit in with a hot fix for 2.5. I'm inclined to want to fix it if we can.

Sounds good to me.

**Comment by dpvc** [comment]
OK, I've fixed the code so that unknown characters are measured even when `noReflows` is true.  This should resolve the problem in the SE posting.

I've done some smoke testing, but didn't rerun the testsuite, as the change was minor.  I did check that it doesn't cause unnecessary reflows.

**Comment by dpvc** [comment]
So I think the branch is ready to go.  Let me know when you do the update, and I'll pull it into `v2.5-latest` and `develop` as well.

**Comment by pkra** [comment]
I've just updated GCS and purged CloudFlare. Changes are coming through on my end.

**Comment by pkra** [comment]
@dpvc I created https://github.com/mathjax/MathJax/milestones/MathJax%20v2.5.1%20hotfix%20release -- did I miss anything that we should add there?

**Comment by dpvc** [comment]
Here's some information from my end related to our talk yesterday about the cache breaking.  I am getting `MathJax.js` v2.5.1, but the copy of `TeX-AMS-MML_HTMLorMML-full.js` is still the v2.5.0 version (it contains none of the 2.5.1 files), even though the browser used  `?rev=2.5.1` to get it.  So It looks like the edge nodes are not asking for the new file due to the CGI parameters.  That only breaks the browser cache (which is what it was originally intended to do).  So purging of the changed files seems also to be necessary.

At this point, however, the browser cache will have the v2.5.0 file associated with the `?rev=2.5.1` since that's what the CDN served when the browser asked for it.  So I'd have to force the refresh to get the new version (when it becomes available).

Just information to keep in mind as we do such updates.

**Comment by pkra** [comment]
Darn. Thanks for checking. I've initiated another purge which hopefully helps.

**Comment by dpvc** [comment]
I'm still getting the 2.5.0 config files from both mathjax/latest and mathjax/2.5-latest.  But I do get the v2.5.1 files if I ask for the individually.  Did you purge the config directory?

**Comment by dpvc** [comment]
The milestone looks correct.

**Comment by dpvc** [comment]
I notices that Math.StackExchange is still using the beta copy.  We might want to suggest that they move back to the official release.

**Comment by pkra** [comment]
OK. I had used an incorrect wildcard while updating GCS (but confusingly it partially did what I wanted; weird).

I've updated GCS and purged Clouflare. Third time's the charm?

**Comment by pkra** [comment]
Forget what I just wrote -- the files are wrong in my local copy. I have no idea what happened.

**Comment by pkra** [comment]
Oh nevermind. They're alright. I just got confused by the TeX input still being at 2.5.0.

**Comment by pkra** [comment]
Are they coming through on your end?

**Comment by dpvc** [comment]
If I use the address without the `rev=2.5.1` I get the correct version.  With the `rev=2.5.1` I get the old version.  This sounds somewhat familiar.  I'm wondering if the "aggressive" caching is the correct setting on the CDN?

**Comment by dpvc** [comment]
I get the correct version for both on the `2.5-latest` branch, but not on the `latest` one.  This should not be this hard!

**Comment by pkra** [comment]
I think I spotted a problem with gsutil -- it's not uploading everything... Investigating.

**Comment by pkra** [comment]
Ok. I checked the files on GCS and it was only the configs that were somehow not uploaded (fun fact: gsutil throws `CommandException: Wrong number of arguments for "cp" command.` on those files. I have no idea why that would be...)

Are all changes coming through on your end now as well?

**Comment by dpvc** [comment]
No, I'm still getting the strange behavior on `mathjax/latest` where I get the new version without `rev=2.5.1` but get the old version with it.  It seems that the `rev=2.5.1` version has been cached by the edge node I'm using from when the old config file wasn't yet replaced.  So now I'm getting that all the time.  That is why I'm wondering if "aggressive" is the right caching setting.  I don't know if there is a way for you to purge that.

I'm wondering if we should remove or change the revision number in MathJax.js so that we avoid those cached copies, now that the files have made it to the CDN properly.

**Comment by pkra** [comment]
Is there a particular file where you get this?

**Comment by dpvc** [comment]
It is [cdnVersion](https://github.com/mathjax/MathJax/blob/master/unpacked/MathJax.js#L50) in MathJax.js.  I would suggest making this something like `2.5.1x`.  It is used only for the `?rev=x.y.z` string, so changing it to something else will mean that none of the cached versions on the CDN will be used.

**Comment by pkra** [comment]
As per our call, I'm refraining for now since the CDN seems to have normalized.
